mne/preprocessing/tests/test_realign.py

```diff
@@ -45,45 +52,87 @@ def test_realign(ratio_other, start_raw, start_other, stop_raw, stop_other):
     other = RawArray(data_other, info_other, first_samp=222)
 
-    # naive processing
     evoked_raw, events_raw, _, events_other = _assert_similarity(raw, other, n_events)
-    if start_raw == start_other:  # can just naively crop
-        a, b = data_raw[0], data_other[0]
-        n = min(len(a), len(b))
-        corr = np.corrcoef(a[:n], b[:n])[0, 1]
-        min_, max_ = (0.99999, 1.0) if sfreq_raw == sfreq_other else (0.8, 0.9)
-        assert min_ <= corr <= max_
+
+    if test_annot:
+        onsets_raw = (events_raw[:, 0] - raw.first_samp) / raw.info["sfreq"]
+        dur_raw = [box_len] * len(onsets_raw)
+        desc_raw = ["raw_box"] * len(onsets_raw)
+        annot_raw = Annotations(onsets_raw, dur_raw, desc_raw)
+        raw.set_annotations(annot_raw)
+
+        onsets_other = (events_other[:, 0] - other.first_samp) / other.info["sfreq"]
+        dur_other = [box_len * ratio_other] * len(onsets_other)
+        desc_other = ["other_box"] * len(onsets_other)
+        annot_other = Annotations(onsets_other, dur_other, desc_other)
+        other.set_annotations(annot_other)
 
     # realign
-    t_raw = (events_raw[:, 0] - raw.first_samp) / other.info["sfreq"]
+    t_raw = (events_raw[:, 0] - raw.first_samp) / raw.info["sfreq"]
     t_other = (events_other[:, 0] - other.first_samp) / other.info["sfreq"]
     assert duration - 10 <= len(events_raw) < duration
     raw_orig, other_orig = raw.copy(), other.copy()
-    realign_raw(raw, other, t_raw, t_other)
+    realign_raw(raw, other, t_raw, t_other, realign_annot=test_annot)
 
     # old events should still work for raw and produce the same result
-    evoked_raw_2, _, _, _ = _assert_similarity(
+    evoked_raw_2, events_raw, _, events_other = _assert_similarity(
         raw, other, n_events, events_raw=events_raw
     )
     assert_allclose(evoked_raw.data, evoked_raw_2.data)
     assert_allclose(raw.times, other.times)
+
     # raw data now aligned
-    corr = np.corrcoef(raw.get_data([0])[0], other.get_data([0])[0])[0, 1]
-    assert 0.99 < corr <= 1.0
+    hann_corr = np.corrcoef(raw.get_data([1])[0], other.get_data([1])[0])[0, 1]
+    assert 0.98 < hann_corr <= 1.0
+    box_corr = np.corrcoef(raw.get_data([2])[0], other.get_data([2])[0])[0, 1]
+    assert 0.98 < box_corr <= 1.0
+
+    # annotations now aligned
+    if test_annot:
+        assert_allclose(
+            raw.annotations.onset, events_raw[:, 0] / raw.info["sfreq"], atol=1 / sfreq
+        )
+        assert_allclose(
+            other.annotations.onset,
+            events_other[:, 0] / other.info["sfreq"],
+            atol=1 / sfreq,
+        )
+        onsets_raw, dur_raw = (
+            raw.annotations.onset - raw.first_samp / sfreq,
+            raw.annotations.duration,
+        )
+        onsets_other, dur_other = (
+            other.annotations.onset - other.first_samp / sfreq,
+            raw.annotations.duration,
+        )
```

Comment:
This is hard to read, and one line per assignment would be shorter anyway. When reformatting it I noticed that the definition of `dur_other` is wrong
```suggestion
        onsets_raw = raw.annotations.onset - raw.first_samp / sfreq
        dur_raw = raw.annotations.duration
        onsets_other = other.annotations.onset - other.first_samp / sfreq
        dur_other = other.annotations.duration
```